tools/data_integrity/fix_candidate_age.php

```diff
@@ -17,62 +17,95 @@
+    try {
+        $instrument = NDB_BVL_Instrument::factory($inst, "", "");
+    } catch (Exception $e) {
+        echo "\t$inst does not seem to be a valid instrument.\n";
+        continue;
+    }
+
+    $table = $instrument->table;
+
+    if (in_array($table, $verifiedTables, true)) {
+        echo "\tTable $table has already been verified. skipping.\n";
+        continue;
+    } else {
+        //Add table name to repertoire to make sure it is not checked again
+        //in case 2+ instruments use the same table
+        $verifiedTables[] = $table;
+    }
+
+    if (!$DB->tableExists($table)) {
+        echo "\tTable $table for instrument $inst does not exist in the Database\n";
         continue;
     } else if (strpos('_proband', $inst) !== false) {
-        echo $inst." is a Proband instrument and should be handled seperately.\n";
+        echo "\t$inst is a Proband instrument and should be handled seperately.\n";
+        continue;
+    }
+
+    // Skip if Date taken does not exist
+    if (!$DB->columnExists($inst, 'Date_taken')) {
+        echo "\t$inst does not use a `Date_taken` field and should be handled separately.\n";
+        continue;
+    }
+    // Skip if Date taken does not exist
+    if (!$DB->columnExists($inst, 'Candidate_Age')) {
+        echo "\t$inst does not use a `Candidate_Age` field and should be handled separately.\n";
         continue;
     }
 
     //Get instrument SQL table
     $DBInstTable = $DB->pselect(
-        "SELECT * 
-         FROM $inst i 
+        "SELECT i.CommentID, Date_taken, Candidate_Age
+         FROM $table i 
             LEFT JOIN flag f ON (i.commentID=f.CommentID)
             LEFT JOIN session s ON (f.SessionID=s.ID)
             LEFT JOIN candidate c ON (s.CandID=c.CandID)
          WHERE c.Active='Y' AND s.Active='Y'",
         array()
     );
 
-    // Break if Date taken does not exist
-    if (!$DB->columnExists($inst, 'Date_taken')) {
-        echo $inst." does not use a `Date_taken` field and should be handled seperately.\n";
-        continue;
-    }
-
     foreach ($DBInstTable as $k => $row) {
-        // Get Instrument Instance
+        // Get Instrument Instance with commentID
         try {
             $instrument = NDB_BVL_Instrument::factory($inst, $row['CommentID'], '', false);
-        } catch (LorisException $e) {
-            echo "$inst instrument row with CommentID: ".$row['CommentID']." was ".
+        } catch (Exception $e) {
+            echo "\t$inst instrument row with CommentID: ".$row['CommentID']." was ".
                 " Ignored for one of the following reasons:\n".
                 "  - The candidate is inactive.\n".
                 "  - The session is inactive.\n\n";
             continue;
         }
+
         if (!$instrument) {
             // instrument does not exist
-            echo $inst." for CommentID:$row[CommentID] could not be instantiated.\n";
+            echo "\t$inst for CommentID:$row[CommentID] could not be instantiated.\n";
```

Comment:
should it be
```suggestion
            echo "\t$inst for CommentID:$row['CommentID'] could not be instantiated.\n";
```
to properly display?

Reply:
Because I'm still inside the string to evaluate. PHP evaluates `"$row[CommentID]"` the same as `$row["CommentID"]`

look for "Example #10 Simple syntax example" in here http://php.net/manual/en/language.types.string.php#language.types.string.parsing.simple